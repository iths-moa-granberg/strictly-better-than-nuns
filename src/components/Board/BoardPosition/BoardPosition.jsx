import React from 'react';

const BoardPosition = ({ position, className, children, clickHandler }) => {
  return (
    <div
      className={className}
      style={{ top: `${position.y}px`, left: `${position.x}px` }}
      onClick={() => clickHandler(position)}>
      <p>{position.id}</p>
      {children}
    </div>
  );
};

export default BoardPosition;
