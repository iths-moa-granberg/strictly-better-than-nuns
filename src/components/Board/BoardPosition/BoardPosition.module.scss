@import '../../../utils';

.position {
  width: 8px;
  height: 8px;

  background-color: $position-background-color;
  border: 2px solid $position-border-color;
  border-radius: 100%;

  position: absolute;
  z-index: 0;

  display: flex;
  justify-content: center;
  align-items: center;
}

.enemy1-end-of-path,
.reachable-e1 {
  background-color: $enemy1-main-color;
}

.enemy2-end-of-path,
.reachable-e2 {
  background-color: $enemy2-main-color;
}

.reachable-0 {
  background-color: $player0-main-color;
}

.reachable-1 {
  background-color: $player1-main-color;
}

.reachable-2 {
  background-color: $player2-main-color;
}

.reachable-3 {
  background-color: $player3-main-color;
}

.reachable-4 {
  background-color: $player4-main-color;
}

.reachable-5 {
  background-color: $player5-main-color;
}
