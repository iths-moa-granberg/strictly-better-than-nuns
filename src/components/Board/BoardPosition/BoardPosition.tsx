import React from 'react';
import { Position } from '../../../shared/sharedTypes';
import styles from './BoardPosition.module.scss';

interface BoardPositionProps {
  readonly position: Position;
  readonly className: {
    reachable: string;
    clickable: string;
  };
  readonly children: JSX.Element[];
  readonly clickHandler: (position: Position) => void;
}

const BoardPosition = ({ position, className, children, clickHandler }: BoardPositionProps) => {
  const multipleChildren = children.length > 1 ? 'multipleChildren' : '';

  return (
    <div
      className={`
        ${styles.position} 
        ${styles[className.reachable]} 
        ${styles[className.clickable]} 
        ${styles[multipleChildren]}
      `}
      style={{ top: `${position.y * (856 / 900)}vh`, left: `${position.x * (856 / 900)}vh` }}
      onClick={() => clickHandler(position)}>
      {children}
    </div>
  );
};

export default BoardPosition;
