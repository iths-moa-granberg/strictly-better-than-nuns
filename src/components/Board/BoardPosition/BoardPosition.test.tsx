import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import BoardPosition from './BoardPosition';

describe('BoardPosition', () => {
  it('renders correctly with one child and no classnames', () => {
    const position = { id: 1, x: 20, y: 30, neighbours: [], inSight: [] };
    const className = { reachable: '', clickable: '' };
    const children = [<div key="1">Child</div>];
    const clickHandler = jest.fn();

    const { container } = render(
      <BoardPosition position={position} className={className} children={children} clickHandler={clickHandler} />
    );

    expect(container).toMatchSnapshot();
  });

  it('renders correctly with one child, reachable & clickable', () => {
    const position = { id: 1, x: 20, y: 30, neighbours: [], inSight: [] };
    const className = { reachable: 'reachable-1', clickable: 'clickable' };
    const children = [<div key="1">Child</div>];
    const clickHandler = jest.fn();

    const { container } = render(
      <BoardPosition position={position} className={className} children={children} clickHandler={clickHandler} />
    );

    expect(container).toMatchSnapshot();
  });

  it('renders correctly with multiple children, reachable & clickable', () => {
    const position = { id: 1, x: 20, y: 30, neighbours: [], inSight: [] };
    const className = { reachable: 'reachable-1', clickable: 'clickable' };
    const children = [<div key="1">Child</div>, <div key="2">Another child</div>];
    const clickHandler = jest.fn();

    const { container } = render(
      <BoardPosition position={position} className={className} children={children} clickHandler={clickHandler} />
    );

    expect(container).toMatchSnapshot();
  });

  it('should send position to clickhandler when clicked', () => {
    const position = { id: 1, x: 20, y: 30, neighbours: [], inSight: [] };
    const className = { reachable: 'reachable-1', clickable: 'clickable' };
    const children = [<div key="1">Child</div>];
    const clickHandler = jest.fn();

    const { container } = render(
      <BoardPosition position={position} className={className} children={children} clickHandler={clickHandler} />
    );

    const div = container.querySelector('div');
    fireEvent.click(div!);

    expect(clickHandler).toHaveBeenCalledWith(position);
  });
});
