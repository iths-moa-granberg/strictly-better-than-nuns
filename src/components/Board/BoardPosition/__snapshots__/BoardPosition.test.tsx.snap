// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardPosition renders correctly with multiple children, reachable & clickable 1`] = `
<div>
  <div
    class="
        position 
        reachable-1 
        clickable 
        multipleChildren
      "
    style="top: 28.533333333333335vh; left: 19.022222222222222vh;"
  >
    <div>
      Child
    </div>
    <div>
      Another child
    </div>
  </div>
</div>
`;

exports[`BoardPosition renders correctly with one child and no classnames 1`] = `
<div>
  <div
    class="
        position 
         
         
        
      "
    style="top: 28.533333333333335vh; left: 19.022222222222222vh;"
  >
    <div>
      Child
    </div>
  </div>
</div>
`;

exports[`BoardPosition renders correctly with one child, reachable & clickable 1`] = `
<div>
  <div
    class="
        position 
        reachable-1 
        clickable 
        
      "
    style="top: 28.533333333333335vh; left: 19.022222222222222vh;"
  >
    <div>
      Child
    </div>
  </div>
</div>
`;
